Firestore/Swift/Source/PropertyWrapper/FirestoreQueryObservable.swift

```diff
@@ -24,19 +24,26 @@ internal class FirestoreQueryObservable<T: Decodable>: ObservableObject {
   private let firestore = Firestore.firestore()
   private var listener: ListenerRegistration? = nil
 
-  init(collectionPath: String, predicates: [QueryPredicate]) {
-    setupListener(from: collectionPath, withPredicates: predicates)
+  internal var configuration: FirestoreQueryConfiguration {
+    didSet {
+      removeListener()
+      setupListener()
+    }
+  }
+
+  init(configuration: FirestoreQueryConfiguration) {
+    self.configuration = configuration
+    setupListener()
   }
 
   deinit {
     removeListener()
   }
 
-  private func setupListener(from collectionPath: String,
-                             withPredicates predicates: [QueryPredicate]) {
-    var query: Query = firestore.collection(collectionPath)
+  private func setupListener() {
```

Comment:
The QueryPredicate and the built in construction of a query seems very powerful. Might that be extracted so it’s possible to use this with the basic firestore api? That can of course come later.